x-pack/plugins/stack_connectors/server/connector_types/inference/index.ts

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import {
+  SubActionConnectorType,
+  ValidatorType,
+} from '@kbn/actions-plugin/server/sub_action_framework/types';
+import {
+  GenerativeAIForSearchPlaygroundConnectorFeatureId,
+  GenerativeAIForSecurityConnectorFeatureId,
+} from '@kbn/actions-plugin/common';
+import { ValidatorServices } from '@kbn/actions-plugin/server/types';
+import { GenerativeAIForObservabilityConnectorFeatureId } from '@kbn/actions-plugin/common/connector_feature_config';
+import { InferenceTaskType } from '@elastic/elasticsearch/lib/api/types';
+import { ElasticsearchClient, Logger } from '@kbn/core/server';
+import {
+  INFERENCE_CONNECTOR_TITLE,
+  INFERENCE_CONNECTOR_ID,
+  ServiceProviderKeys,
+  SUB_ACTION,
+} from '../../../common/inference/constants';
+import { ConfigSchema, SecretsSchema } from '../../../common/inference/schema';
+import { Config, Secrets } from '../../../common/inference/types';
+import { InferenceConnector } from './inference';
+import { unflattenObject } from '../lib/unflatten_object';
+
+const deleteInferenceEndpoint = async (
+  inferenceId: string,
+  taskType: InferenceTaskType,
+  logger: Logger,
+  esClient: ElasticsearchClient
+) => {
+  try {
+    await esClient?.inference.delete({
+      task_type: taskType,
+      inference_id: inferenceId,
+    });
+    logger.info(
+      `Inference endpoint for task type "${taskType}" and inference id ${inferenceId} was successfuly deleted`
+    );
+  } catch (e) {
+    logger.warn(
+      `Failed to delete inference endpoint for task type "${taskType}" and inference id ${inferenceId}. Error: ${e.message}`
+    );
+    throw e;
+  }
+};
+
+export const getConnectorType = (): SubActionConnectorType<Config, Secrets> => ({
+  id: INFERENCE_CONNECTOR_ID,
+  name: INFERENCE_CONNECTOR_TITLE,
+  getService: (params) => new InferenceConnector(params),
+  schema: {
+    config: ConfigSchema,
+    secrets: SecretsSchema,
+  },
+  validators: [{ type: ValidatorType.CONFIG, validator: configValidator }],
+  supportedFeatureIds: [
+    GenerativeAIForSecurityConnectorFeatureId,
+    GenerativeAIForSearchPlaygroundConnectorFeatureId,
+    GenerativeAIForObservabilityConnectorFeatureId,
+  ],
+  minimumLicenseRequired: 'enterprise' as const,
+  preSaveHook: async ({ config, secrets, logger, services, isUpdate }) => {
+    const esClient = services.scopedClusterClient.asCurrentUser;
+    try {
+      const taskSettings = config?.taskTypeConfig
+        ? {
+            ...unflattenObject(config?.taskTypeConfig),
+          }
+        : {};
+      const serviceSettings = {
+        ...unflattenObject(config?.providerConfig ?? {}),
+        ...unflattenObject(secrets?.providerSecrets ?? {}),
```

Comment:
No, and it shouldn't. Because when we are creating the pre-configured connector we suppose to have already the endpoint, also it's impossible to edit and delete pre-configured connector, all operations are manul.

Reply:
Ya, what Yuliia said is correct - I have no idea if not having pre-defined connectors will be a problem for someone.  

I think the problem from a response-ops framework side is that this is apparently a connector type that you should not pre-define, and we don’t (currently) have a way of preventing that.  

I guess it depends on how obvious the connector is broken in the UX, or responds to execution requests - we'd probably want to indicate in error notifications that these can't be pre-defined.  Could also log something at startup if they are found.

Reply:
So, I don't have all use cases of preconfigured connectors in mind, but it feels strange to me. I know we're massively relying on preconfigured connectors for FTR tests for examples, and this means that we won't be able to use preconfigured inference connectors for tests. 

But that's not my area of expertise, I'll let someone from @elastic/response-ops decide on that point.

Reply:
I wouldn't say we're **massively** relying on pre-configured connectors for FTR - here's our "common" FTR config: https://github.com/elastic/kibana/blob/eefabb0f534234d6c2c0e3468bbdc65a16009e93/x-pack/test/alerting_api_integration/common/config.ts#L228-L343

I think those are there for ease-of-use, more than anything else.  I don't know of any reason why you can't create a connector in FTR via API, and of course, we should be testing that (and are, but I'm not 100% about all the connector types).

If we absolutely need it in the future, I can imagine some ways it could be done.  We may need to persist some info about those connectors, which we don't do now - they exist only in memory.

Reply:
Opened an issue to look into "disabling" preconfigured connectors that use hooks: https://github.com/elastic/kibana/issues/195824 - guessing it's not required and we'd be backlogging it, but also seems fairly straight-forward to do ...